src/implementation.cpp

```diff
@@ -100,7 +100,7 @@ class unsupported_implementation final : public implementation {
   unsupported_implementation() : implementation("unsupported", "Unsupported CPU (no detected SIMD instructions)", 0) {}
 };
 
-const unsupported_implementation unsupported_singleton{};
+static const unsupported_implementation unsupported_singleton{};
```

Comment:
I think we want there only to be one of these--therefore it should not be static. With static, the compiler will not warn you if you try to make multiple copies, but with const it will. The reason we want one is we're going to be taking the address of it and I wouldn't mind the option of being able to *compare* the address.

Reply:
Same goes for the others in this file.